test/integration/BCowPool.t.sol

```diff
@@ -2,87 +2,67 @@
 
-contract BCowPoolIntegrationTest is Test, BCoWConst, BMath, GasSnapshot {
+contract BCowPoolIntegrationTest is PoolSwapIntegrationTest, BCoWConst {
   using GPv2Order for GPv2Order.Data;
 
-  IBCoWPool public pool;
-
-  IERC20 public dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
-  IERC20 public weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
-  ISettlement public settlement = ISettlement(0x9008D19f58AAbD9eD0D60971565AA8510560ab41);
   address public solver = address(0xa5559C2E1302c5Ce82582A6b1E4Aec562C2FbCf4);
 
-  address public controller = makeAddr('controller');
-  Vm.Wallet swapper = vm.createWallet('swapper');
+  ISettlement public settlement = ISettlement(0x9008D19f58AAbD9eD0D60971565AA8510560ab41);
 
   bytes32 public constant APP_DATA = bytes32('exampleIntegrationAppData');
 
-  uint256 public constant HUNDRED_UNITS = 100 ether;
-  uint256 public constant ONE_UNIT = 1 ether;
-
-  function setUp() public {
+  function setUp() public override {
     vm.createSelectFork('mainnet', 20_012_063);
 
-    // deal controller
-    deal(address(dai), controller, HUNDRED_UNITS);
-    deal(address(weth), controller, HUNDRED_UNITS);
-
-    // deal swapper
-    deal(address(weth), swapper.addr, HUNDRED_UNITS);
-
-    vm.startPrank(controller);
-    // deploy
-    // TODO: deploy with BCoWFactory
-    pool = new BCoWPool(address(settlement));
-    // bind
-    dai.approve(address(pool), type(uint256).max);
-    weth.approve(address(pool), type(uint256).max);
-    IBPool(pool).bind(address(dai), HUNDRED_UNITS, 2e18);
-    IBPool(pool).bind(address(weth), HUNDRED_UNITS, 8e18);
+    deal(address(dai), lp, HUNDRED_UNITS);
+    deal(address(weth), lp, HUNDRED_UNITS);
+
+    deal(address(dai), swapper.addr, ONE_UNIT);
+
+    vm.startPrank(lp);
+    // TODO: deploy using BCoWFactory
+    pool = address(new BCoWPool(address(settlement)));
+
+    dai.approve(pool, type(uint256).max);
+    weth.approve(pool, type(uint256).max);
+    IBCoWPool(pool).bind(address(dai), ONE_UNIT, 2e18); // 20% weight
+    IBCoWPool(pool).bind(address(weth), ONE_UNIT, 8e18); // 80% weight
     // finalize
-    IBPool(pool).finalize();
+    IBCoWPool(pool).finalize();
     // enable trading
-    pool.enableTrading(APP_DATA);
-  }
+    IBCoWPool(pool).enableTrading(APP_DATA);
 
-  function testBCowPoolSwap() public {
-    uint256 buyAmount = HUNDRED_UNITS;
-    uint256 sellAmount = calcOutGivenIn({
-      tokenBalanceIn: dai.balanceOf(address(pool)),
-      tokenWeightIn: pool.getDenormalizedWeight(address(dai)),
-      tokenBalanceOut: weth.balanceOf(address(pool)),
-      tokenWeightOut: pool.getDenormalizedWeight(address(weth)),
-      tokenAmountIn: buyAmount,
-      swapFee: 0
-    });
+    // clean dai and weth from the settlement
+    vm.startPrank(address(settlement));
+    dai.transfer(address(0), dai.balanceOf(address(settlement)));
+    weth.transfer(address(0), weth.balanceOf(address(settlement)));
```

Comment:
is this needed? do we check `settlement` balances?

Reply:
no but the settlement has some dust in DAI and WETH, and dont want it to mess with the tests, just preventive

Reply:
hmmm ok but i think is more realistic if we don't do this, anyways, it should be indistinct for the scope of these tests (bc we're not checking settlement balances ever)